packages/client/lib/rpc/modules/debug.ts

```diff
@@ -0,0 +1,122 @@
+import { bigIntToHex, bufferToHex, toBuffer } from '@ethereumjs/util'
+
+import { INTERNAL_ERROR, INVALID_PARAMS } from '../error-code'
+import { middleware, validators } from '../validation'
+
+import type { EthereumClient } from '../..'
+import type { FullEthereumService } from '../../service'
+
+export interface tracerOpts {
+  disableStack?: boolean
+  disableStorage?: boolean
+  enableMemory?: boolean
+  enableReturnData?: boolean
+}
+/**
+ * web3_* RPC module
+ * @memberof module:rpc/modules
+ */
+export class Debug {
+  private service: FullEthereumService
+  /**
+   * Create debug_* RPC module
+   * @param client Client to which the module binds
+   */
+  constructor(client: EthereumClient) {
+    this.service = client.services.find((s) => s.name === 'eth') as FullEthereumService
+    this.traceTransaction = middleware(this.traceTransaction.bind(this), 1, [[validators.hex]])
+  }
+
+  /**
+   * Returns a call trace for the requested transaction or null if not available
+   * @param params string representing the transaction hash
+   */
+  async traceTransaction(params: [string, tracerOpts]) {
+    const [txHash, opts] = params
+
+    // Validate configuration and parameters
+    if (!this.service.execution.receiptsManager) {
+      throw {
+        message: 'missing receiptsManager',
+        code: INTERNAL_ERROR,
+      }
+    }
+    if (opts.disableStorage === false) {
+      throw {
+        code: INVALID_PARAMS,
+        message: 'storage retrieval not implemented',
+      }
+    }
+
+    try {
+      const result = await this.service.execution.receiptsManager.getReceiptByTxHash(
+        toBuffer(txHash)
+      )
+      if (!result) return null
+      const [_, blockHash, txIndex] = result
+      const block = await this.service.chain.getBlock(blockHash)
+      const parentBlock = await this.service.chain.getBlock(block.header.parentHash)
+      const tx = block.transactions[txIndex]
+
+      // Copy VM so as to not modify state when running transactions being traced
+      const vmCopy = await this.service.execution.vm.copy()
+      await vmCopy.stateManager.setStateRoot(parentBlock.header.stateRoot)
+      for (let x = 0; x < txIndex; x++) {
+        // Run all txns in the block prior to the traced transaction
+        await vmCopy.runTx({ tx: block.transactions[x], block })
+      }
+
+      const trace: any = {
+        gas: '',
+        returnValue: '',
+        failed: false,
+        structLogs: [],
+      }
```

Comment:
Nice! The `trace: any` can then be removed right?

Reply:
No, I guess I can add it but not sure how much value it adds since this gets converted directly to json.

Reply:
It would for example enable autocompletion which we lose when typecasting as any. 

From checking locally the only type error we get when remove the :any is this line:
`trace.structLogs[trace.structLogs.length - 1].error = true` 

Can't we just adjust the structLog by adding this prop:
`error: boolean | undefined | null` ?